### PR TITLE
ref(dashboards): Move search bar to config

### PR DESCRIPTION
**Pull request by shruthilayaj**
Add search bar to config

### CODE REVIEW

File: static/app/views/dashboardsV2/widgetBuilder/buildSteps/filterResultsStep/issuesSearchBar.tsx [hunk 1]
```diff
@@ -19,29 +19,27 @@ interface Props {
...
-  selection,
-  searchSource,
+  pageFilters,
 }: Props) {
   const api = useApi();
   function tagValueLoader(key: string, search: string) {
-    const orgId = organization.slug;
-    const projectIds = selection.projects.map(id => id.toString());
+    const orgId = organization!.slug;
+    const projectIds = pageFilters?.projects.map(id => id.toString());
```
Comment (narsaynorath) [inline static/app/views/dashboardsV2/widgetBuilder/buildSteps/filterResultsStep/issuesSearchBar.tsx:38]: `projects` and `datetime` aren't optional in `pageFilters` according to the type, can we drop `?`?

**Review by narsaynorath — COMMENTED** [review]
Mostly small comments, but a bigger one regarding how we want to call things that return components

File: static/app/views/dashboardsV2/widgetBuilder/buildSteps/filterResultsStep/releaseSearchBar.tsx [hunk 2]
```diff
@@ -77,7 +86,7 @@ export function ReleaseSearchBar({orgSlug, query, projectIds, onSearch, onBlur}:
           maxQueryLength={MAX_QUERY_LENGTH}
           maxSearchItems={MAX_SEARCH_ITEMS}
           searchSource="widget_builder"
-          query={query.conditions}
+          query={widgetQuery.conditions}
```
Comment (narsaynorath) [inline static/app/views/dashboardsV2/widgetBuilder/buildSteps/filterResultsStep/releaseSearchBar.tsx:89]: We only seem to use `widgetQuery` to get `conditions` from it. What do you think about just passing down `conditions`?

Comment (shruthilayaj) [reply static/app/views/dashboardsV2/widgetBuilder/buildSteps/filterResultsStep/releaseSearchBar.tsx:89]: eh I think it's fine to pass the widget query for now, it's a nice pattern that matches the rest of the config. we can clean it up in a follow up if necessary

File: static/app/views/dashboardsV2/widgetBuilder/buildSteps/filterResultsStep/index.tsx [hunk 3]
```diff
@@ -113,32 +111,13 @@ export function FilterResultsStep({
...
-                    onSearch={handleSearch(queryIndex)}
-                  />
-                )}
+                {datasetConfig.getSearchBar({
+                  organization,
+                  pageFilters: selection,
+                  onBlur: handleBlur(queryIndex),
+                  onSearch: handleSearch(queryIndex),
+                  widgetQuery: query,
+                })}
```
[the remark this answers is not in the document]
Comment (shruthilayaj) [reply static/app/views/dashboardsV2/widgetBuilder/buildSteps/filterResultsStep/index.tsx]: love this suggestion!

**Review by narsaynorath — APPROVED** [review]
Awesome refactor :fire:

**Pull request merged**